simtools/corsika/corsika_config.py

```diff
@@ -0,0 +1,522 @@
+    layout: LayoutArray
+        Instance of LayoutArray.
+    label: str
+        Instance label.
+    primary: str
+        Name of the primary particle (e.g gamma, proton ...).
+
+    Methods
+    -------
+    setUserParameters(corsikaConfigData)
+        Set user parameters from a dict.
+    getUserParameter(parName)
+        Get the value of a user parameter.
+    printUserParameters()
+        Print user parameters for inspection.
+    getOutputFileName(runNumber)
+        Get the name of the CORSIKA output file for a certain run.
+    exportInputFile()
+        Create and export CORSIKA input file.
+    getInputFile()
+        Get the full path of the CORSIKA input file.
+    getInputFileNameForRun(runNumber)
+        Get the CORSIKA input file for one specific run.
+        This is the input file after being pre-processed by sim_telarray (pfp).
+    '''
+
+    def __init__(
+        self,
+        site,
+        layoutName,
+        label=None,
+        filesLocation=None,
+        corsikaConfigData=None,
+        corsikaConfigFile=None,
+        corsikaParametersFile=None
+    ):
+        '''
+        CorsikaConfig init.
+
+        Parameters
+        ----------
+        site: str
+            South or North.
+        layoutName: str
+            Name of the layout.
+        layout: LayoutArray
+            Instance of LayoutArray.
+        label: str
+            Instance label.
+        filesLocation: str or Path.
+            Main location of the output files.
+        corsikaConfigData: dict
+            Dict with CORSIKA config data.
+        corsikaConfigFile: str or Path
+            Path to yaml file containing CORSIKA config data.
+        corsikaParametersFile: str or Path
+            Path to yaml file containing CORSIKA parameters.
+        '''
+
+        self._logger = logging.getLogger(__name__)
+        self._logger.debug('Init CorsikaConfig')
+
+        self.label = label
+        self.site = names.validateSiteName(site)
+
+        self._filesLocation = cfg.getConfigArg('outputLocation', filesLocation)
+
+        # Grabbing layout name and building LayoutArray
+        self.layoutName = names.validateLayoutArrayName(layoutName)
+        self.layout = LayoutArray.fromLayoutArrayName(
+            self.site + '-' + self.layoutName,
+            label=self.label
+        )
+
+        # Load parameters
+        self._loadCorsikaParametersFile(corsikaParametersFile)
+
+        corsikaConfigData = collectDataFromYamlOrDict(corsikaConfigFile, corsikaConfigData)
+        self.setUserParameters(corsikaConfigData)
+        self._isFileUpdated = False
+
+    def __repr__(self):
+        text = '<class {}> (site={}, layout={}, label={})'.format(
+            self.__class__.__name__,
+            self.site,
+            self.layoutName,
+            self.label
+        )
+        return text
+
+    def _loadCorsikaParametersFile(self, filename):
+        '''
+        Load CORSIKA parameters from a file, if given (filename not None),
```

Comment:
I think we are loading on both cases from a file, maybe write the commend a little bit different that this is clearer.

Reply:
That is actually written in the docstring.
I changed to '... from a given file ...' to make it clearer.

Reply:
OK!